From 2db1d1ba1e3473df2bac643632cae1b077a838b5 Mon Sep 17 00:00:00 2001
From: Takatoshi Akiyama <takatoshi.akiyama.kj@ps.hitachi-solutions.com>
Date: Thu, 16 Nov 2017 16:34:13 +0900
Subject: [PATCH 173/909] DT: hwspinlock: Add binding documentation for R-Car
 Gen3 hwspinlock

commit cff1afec5e86e03d61a82726071b272ca257c9c8 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Add binding documentation for the R-Car Gen3 Hardware Spinlock.

Signed-off-by: Takatoshi Akiyama <takatoshi.akiyama.kj@ps.hitachi-solutions.com>
Signed-off-by: Hiroki Negishi <hiroki.negishi.bx@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../bindings/hwlock/rcar-hwspinlock.txt       | 25 +++++++++++++++++++
 1 file changed, 25 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/hwlock/rcar-hwspinlock.txt

diff --git a/Documentation/devicetree/bindings/hwlock/rcar-hwspinlock.txt b/Documentation/devicetree/bindings/hwlock/rcar-hwspinlock.txt
new file mode 100644
index 000000000000..e5bab0e32cd7
--- /dev/null
+++ b/Documentation/devicetree/bindings/hwlock/rcar-hwspinlock.txt
@@ -0,0 +1,25 @@
+DT bindings for the Renesas R-Car Hardware spinlock driver
+----------------------------------------------------------
+
+Required properties :
+- compatible : shall contain only one of the following:
+	- "renesas,mfis-lock-r8a7795", "renesas,mfis-lock";
+	- "renesas,mfis-lock-r8a7796", "renesas,mfis-lock";
+	- "renesas,mfis-lock-r8a77965", "renesas,mfis-lock";
+
+- reg : start address and length for MFIS registers.
+
+- #hwlock-cells : should be 1. hwspinlock users specify the hwspinlock id
+	to request a specific hwspinlock.
+
+Please look at the generic hwlock binding for usage information for consumers,
+"Documentation/devicetree/bindings/hwlock/hwlock.txt"
+
+Examples:
+
+mfis: mfis@e6260000 {
+	compatible = "renesas,mfis-lock-r8a7795",
+		     "renesas,mfis-lock";
+	reg = <0 0xe6260000 0 0x1000>;
+	#hwlock-cells = <1>;
+};
-- 
2.17.1

